packages/react-server/src/ReactServerStreamConfigBrowser.js

```diff
@@ -21,24 +21,64 @@ export function flushBuffered(destination: Destination) {
   // transform streams. https://github.com/whatwg/streams/issues/960
 }
 
+let currentView = null;
+let writtenBytes = 0;
+
 export function beginWriting(destination: Destination) {}
 
 export function writeChunk(
   destination: Destination,
   chunk: PrecomputedChunk | Chunk,
 ): void {
-  destination.enqueue(chunk);
+  if (currentView === null) {
+    currentView = new Uint8Array(512);
+    writtenBytes = 0;
+  }
+
+  if (chunk.length > currentView.length) {
+    // this chunk is larger than our view which implies it was not
+    // one that is cached by the streaming renderer. We will enqueu
+    // it directly and expect it is not re-used
+    if (writtenBytes > 0) {
+      destination.enqueue(new Uint8Array(currentView.buffer, 0, writtenBytes));
+      currentView = null;
```

Comment:
Ah, maybe because of this case. It is very likely that we're going to have at least one more call after this. Like we have to write `</html>` for example. So you can eagerly allocate the next buffer.

Reply:
One case where this might not hold is if we start using `byobRequest.view` to get the buffer in `beginWriting`. Then we wouldn't need to allocate one for the future.

Reply:
I’m confused, why would using byobRequest make it so we wouldn’t need to allocate a new buffer? What about byob is distinct from our self created buffers that would change this logic?